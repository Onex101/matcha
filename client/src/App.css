.App {
  /* margin-top: 15px; */
  /* background-image: url("./containers/imgs/background.jpg");
  background-size: cover;
  overflow: hidden; */
  
}

.App .navbar-brand {
  font-weight: bold;
}
.App-setup {
  /* padding: 15px; */
  /* background-image: url("./containers/imgs/background.jpg"); */
  /* background-size: cover; */
  /* background-position: center; */
  /* overflow: hidden; */

}

.App-container {
padding-right: 0;
padding-left: 0;
margin-right: auto;
margin-left: auto;
}

.App-logo {
  float: left;
    height: 50px;
    padding: 15px 15px;
    font-size: 18px;
    line-height: 20px;
    font-weight:bold
}

.nav-icon {
  width: 25px;
}

.App-content{
  padding-left: 5px;
  padding-right: 5px;
  overflow: scroll;
  height: 90vh;
}








/* .App .navbar-brand {
  font-weight: bold;
}
.nav-top {
  margin-bottom: 0px !important;
}
.nav-bottom {
  position: absolute;
  bottom: 0;
  width: 100%;
  margin-bottom: 0px !important;
  min-height: 0 !important;
} */

/* .App-container {
  padding-right: 0;
  padding-left: 0;
  margin-right: auto;
  margin-left: auto;
  padding-bottom: 1px;
}

.App-logo {
  float: left;
    height: 50px;
    padding: 15px 15px;
    font-size: 18px;
    line-height: 20px;
    font-weight:bold
}

.nav-icon {
  width: 25px;
}

.App-content{
  padding-left: 5px;
  padding-right: 5px;
  overflow-y: scroll;
  height: 90vh;
} */