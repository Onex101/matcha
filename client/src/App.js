import React, { Component, Fragment } from "react";
import { Link } from "react-router-dom";
import { Nav, Navbar, NavItem } from "react-bootstrap";
import "./App.css";
import Routes from "./Routes";
import { LinkContainer } from "react-router-bootstrap";

class App extends Component {
  constructor(props) {
    super(props);
  
    this.state = {
      isAuthenticated   : false,
      userInfo              : null
    };
  }
  
  userHasAuthenticated = authenticated => {
    this.setState({ isAuthenticated: authenticated });
  }

  setUser = user => {
    this.setState({ userInfo: user });
    console.log(user.data);
    console.log(user.data.user_name);
  }

  handleLogout = event => {
    this.userHasAuthenticated(false);
  }

  render() {
    const childProps = {
      isAuthenticated       : this.state.isAuthenticated,
      userHasAuthenticated  : this.userHasAuthenticated,
      userInfo                  : this.state.userInfo,
      setUser               : this.setUser
    };
    
    return (
      <div className="App container">
        <Navbar fluid collapseOnSelect>
          <Navbar.Header>
            <Navbar.Brand>
              <Link to="/">Matcha</Link>
            </Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
          <Navbar.Collapse>
            <Nav pullRight>
              {this.state.isAuthenticated
                // ? <NavItem onClick={this.handleLogout}>Logout</NavItem>
                ? <Fragment>
                    <LinkContainer to="/signup">
                      <NavItem>{this.state.user.data.user_name}</NavItem>
                      <NavItem>Username</NavItem>
                    </LinkContainer>
                    {/* <LinkContainer to="/login"> */}
                    <NavItem onClick={this.handleLogout}>Logout</NavItem>
                    {/* </LinkContainer> */}
                  </Fragment>
                : <Fragment>
                    <LinkContainer to="/signup">
                      <NavItem>Signup</NavItem>
                    </LinkContainer>
                    <LinkContainer to="/login">
                      <NavItem>Login</NavItem>
                    </LinkContainer>
                  </Fragment>
              }
            </Nav>
          </Navbar.Collapse>
        </Navbar>
        <Routes childProps={childProps} />
      </div>
    );
  }  
}

export default App;
