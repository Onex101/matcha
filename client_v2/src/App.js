import React from 'react';
import logo from './logo.svg';
import './App.css';
import Routes from "./Routes";

function App() {
  const childProps = {
    // isAuthenticated       : this.state.isAuthenticated,
    // userHasAuthenticated  : this.userHasAuthenticated,
    // userInfo              : this.state.userInfo,
    // userProfile           : this.state.userProfile,
    // userMatches           : this.state.userMatches,
    // setUser               : this.setUser,
    // socket                : this.state.socket,
    // socketUser            : this.state.socketUser,
    // getSearchResults      : this.getSearchResults,
    // gotResults            : this.gotResults,
    // results               : this.state.results,
    // resultType            : this.state.resultType,
    // getMatches            : this.getMatches
  };
  // var style = {
  //   display: 'flex',
  // }

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
      <div className="App-content">
          <Routes childProps={childProps} />
        </div>
    </div>
    
  );
}

export default App;


//  <div className="App-setup">
//       <div className="App-container">
//         <Navbar fluid collapseOnSelect>
//           <Navbar.Header>
//             <Navbar.Brand>
//               <Link to="/" className="App-logo">Matcha</Link>
//             </Navbar.Brand>
//             <div style={style}>
//               {this.searchCheck()}
//               <Navbar.Toggle />
//             </div>
//           </Navbar.Header>
//           {localStorage.getItem('user')
//             ? this.renderUser()
//             : <Navbar.Collapse><Nav pullRight>
//               <Fragment>
//                 <LinkContainer to="/signup">
//                   <NavItem>Signup</NavItem>
//                 </LinkContainer>
//                 <LinkContainer to="/login">
//                   <NavItem>Login</NavItem>
//                 </LinkContainer>
//               </Fragment>
//             </Nav></Navbar.Collapse>
//           }
//         </Navbar>
//         {console.log("Childprops: ")}
//         {console.log("Test for re-render")}
//         {console.info(childProps)}
//         {this.state.redirect ?
//           <Redirect push to="/searchResults" /> : null}
//         <div className="App-content">
//           <Routes childProps={childProps} />
//         </div>
//       </div>
//     </div> 