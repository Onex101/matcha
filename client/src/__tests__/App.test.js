import React from 'react';
import {render, cleanup, fireEvent, waitForElement} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../App.js';

afterEach(cleanup);

it("renders", ()=>{
	const { asFragment,  } = render(<Router>
		<App />
	</Router>);
	expect(asFragment()).toMatchSnapshot();

});