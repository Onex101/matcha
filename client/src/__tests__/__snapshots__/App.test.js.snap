// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="App-setup"
  >
    <div
      class="App-container"
    >
      <nav
        class="nav-top navbar navbar-default"
      >
        <div
          class="container-fluid"
        >
          <div
            class="navbar-header"
          >
            <a
              class="App-logo navbar-brand"
              href="/"
            >
              Matcha
            </a>
            <button
              class="navbar-toggle collapsed"
              type="button"
            >
              <span
                class="sr-only"
              >
                Toggle navigation
              </span>
              <span
                class="icon-bar"
              />
              <span
                class="icon-bar"
              />
              <span
                class="icon-bar"
              />
            </button>
          </div>
          <div
            class="navbar-collapse collapse"
          >
            <ul
              class="nav navbar-nav navbar-right"
            >
              <li
                class=""
                role="presentation"
              >
                <a
                  href="/signup"
                >
                  Signup
                </a>
              </li>
              <li
                class=""
                role="presentation"
              >
                <a
                  href="/login"
                >
                  Login
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        class="App-content"
      >
        <div
          class="Home"
        >
          <div
            class="lander"
          >
            <div>
              <h1>
                Matcha
              </h1>
              <p>
                Find your Match and let Sparks fly!
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fixed-bottom"
      >
        <nav
          class="nav-bottom navbar navbar-default"
          color="dark"
        >
          <div
            class="container"
          >
            <div
              class="navbar-brand"
            >
              <img
                alt=""
                class="nav-icon"
                id="Foot"
                src="footer.png"
              />
            </div>
          </div>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
